src/vtl.ts

```diff
@@ -468,9 +469,10 @@ export abstract class VTL {
           const [previousValue, currentValue, currentIndex, array] =
             fn.parameters;
 
-          const currentValueName = isIdentifier(currentValue.name)
-            ? `$${currentValue.name.name}`
-            : this.newLocalVarName();
+          const currentValueName =
+            currentValue && isIdentifier(currentValue.name)
+              ? `$${currentValue.name.name}`
+              : this.newLocalVarName();
```

Comment:
Could this be simplified as `isIdentifier(currentValue?.name)`

Reply:
:/
![image](https://user-images.githubusercontent.com/289213/183417480-7f5c517b-bb66-4f8b-86dd-161058123b0b.png)

Reply:
Strange, it doesn't narrow it to defined?

Reply:
nope...